trace2/tr2_dst.c

```diff
@@ -57,37 +56,103 @@ static int tr2_dst_try_path(struct tr2_dst *dst, const char *tgt_value)
 
 #ifndef NO_UNIX_SOCKETS
 #define PREFIX_AF_UNIX "af_unix:"
-#define PREFIX_AF_UNIX_LEN (8)
+#define PREFIX_AF_UNIX_STREAM "af_unix:stream:"
+#define PREFIX_AF_UNIX_DGRAM "af_unix:dgram:"
+
+static int tr2_dst_try_uds_connect(const char *path, int sock_type, int *out_fd)
+{
+	int fd;
+	struct sockaddr_un sa;
+
+	fd = socket(AF_UNIX, sock_type, 0);
+	if (fd == -1)
+		return errno;
+
+	sa.sun_family = AF_UNIX;
+	strlcpy(sa.sun_path, path, sizeof(sa.sun_path));
+
+	if (connect(fd, (struct sockaddr *)&sa, sizeof(sa)) == -1) {
+		int e = errno;
+		close(fd);
+		return e;
+	}
+
+	*out_fd = fd;
+	return 0;
+}
+
+#define TR2_DST_UDS_TRY_STREAM (1 << 0)
+#define TR2_DST_UDS_TRY_DGRAM  (1 << 1)
```

Comment:
I do not quite understand how these changes relate to the commit message. Were these changes intended for another commit?

Reply:
Yeah, I need to re-do the port of the trace2 changes from upstream.  After I did this port, I realized that there must be another round of upstream changes that need to be pulled over before this system-config set is applied.

I'm about to push another version of the system-config changes upstream, so I'll revisit this shortly.